### ISSUE TITLE
Code Cleanup | Open to all

### ISSUE THREAD

**Issue by TreyWW** [TreyWW/MyFinances#160]
Hey.

This project has grew, quite a lot, since I first started. It's taken slightly longer than I was expecting and I've in general made it more complex than it needed to be. Nevertheless - it continues!

But in order for this project to be approachable and maintainable, it needs a bit of a tidy up. Anyone that would like to help contribute, feel free to leave a comment here and make a PR when done. You don't need to clean up everything, there's no minimum or maximum limit. Clean up one function, or a file, or a html div, it's up to you! 

This will be open for a while to help boost project approachability and for newcomers to get a better understanding of the projects codebase. 

Thanks all,
I appreciate any help given

**Comment by MuhammadHasnain77** [comment]
can I work  on manage.py file I think it needs some improvement

**Comment by TreyWW** [comment]
Hi @MuhammadHasnain77, since manage.py is a Django generated file I think it's okay to be left how it is. It's never going to be edited so I see no need for any improvements. And I don't want any functionality of it to be accidentally changed as it's the main file to run tasks.

**Comment by MuhammadHasnain77** [comment]
ok

**Comment by TreyWW** [comment]
Feel free to cleanup anything else though. There's python files in /backend!

**Comment by MuhammadHasnain77** [comment]
I will look to them

**Comment by jaypee15** [comment]
Hi, I'm looking to clean up some of the views, will submit a PR when done.

**Comment by rlanier-webdev** [comment]
Hey there @TreyWW. I'd like to help out. Should I list what I'm working on or just submit a PR when done?

**Comment by TreyWW** [comment]
Hi @rlanier-webdev. Up to you. Personally I'd wait as when I merge #87 in a few days that will have about 30 file changes.

We've also started using djlint for html files which is nice

If you do want to work on files, preferably work on like invoices or receipts until #87 merges if that's okay. 

Sorry for the disruption

**Comment by rlanier-webdev** [comment]
No worries. I'll standby. Thanks!

**Comment by adnangif** [comment]
Hi! I would like to contribute to some cleanup/refactoring. I would like to refactor codes in /backend. Is there anything that I should know before I start? Thank you.

**Comment by TreyWW** [comment]
Hey @adnangif,

Not really. Make sure no functionality is changed that's too major, or make sure to document it in your PR if you do change anything. Once done make sure to run the below commands:

```bash
py manage.py test # this runs tests to make sure nothing has broke in the process
black ./ # this runs the black formatter that formats the files
```

Let me know if you've got any questions or need help as you go along. Thanks for your interest!

**Comment by adnangif** [comment]
Okay I will. Thank you again.

**Comment by NikolaSL00** [comment]
I've noticed that all the database models are written in a single lengthy file. Would it be acceptable for me to reorganize them by creating a 'models/' folder and separating each database model into its own file? I'll also ensure to update the corresponding imports accordingly.

**Comment by TreyWW** [comment]
Hey @NikolaSL00,

I guess we could move them to their own files, it just (in my opinion) makes it harder to find and navigate models. I've kind of put this off cause we don't have too many models, I simply collapse each one to only see the name until I need it. 
With their own module it'll have like 10 separate files making it a little harder to find each one. 

We can try it though, feel free to move them into their own modules and we can see how it looks. Thanks :)

**Comment by Domejko** [comment]
Hey,

I would like to help with /backend files. There is couple of nested and repetitive  `if` statements that can be reduced and few `except` blocks that are to broad.

**Comment by TreyWW** [comment]
Hey @Domejko,

That would be great! I agree, there are definitely quite a few nested statements that can be better arranged.

Appreciate it! Let me know if you struggle with anything

**Comment by nsr7t** [comment]
Is there any code cleanup that I can contribute to?

**Comment by FilipeSobrinho** [comment]
Hey,
I've talked with @NikolaSL00 and I will try to do what he mentioned in his comment.

**Comment by vishalprashant** [comment]
Hi,
I am trying to contribute to an open source project for the first time to gain some experience and I have only ever worked in a classroom setting, so I wanted to start on a task like this. I am unfamiliar with the .env file and settings. How much of this do I need to fill out in order to be able to make a simple code refactor/cleanup and make a PR?

**Comment by vishalprashant** [comment]
Also are there always a million dependencies to install even after running the poetry installation script?

**Comment by TreyWW** [comment]
Hey @vishalprashant,
So the env file you can actually leave how it is. You can follow the documentation and setup SQLite and that’s the only real env variable you need. The docs will show you the rest

**Comment by vishalprashant** [comment]
@TreyWW Thanks for the help. You mentioned blank = True at the top. In cases like the Invoice class in models.py why is blank = True even necessary? How does the functionality change if I just remove it from each variable and leave null = True to dictate what needs to be in those fields? Are all these fields not supposed to be mandatory?

**Comment by devenderbutani21** [comment]
Hello! I would love to contribute to the cleanup and refactoring task. Let me know if I need to know anything before starting.

**Comment by TreyWW** [comment]
Hey @devenderbutani21! One key thing I'd suggest is please, please, make smaller PRs rather than bigger ones. They often slow the whole process down, can take weeks and just clog things.

If you'd like to make drastic changes I'd prefer if you could ask first, so you don't waste time for me to just decline it.

Otherwise, make a few changes per PR, i'd rather more PRs but smaller, I can merge them in minutes rather than days.

Thanks! To get started you can use the [docs](https://docs.strelix.org/MyFinances/)

**Comment by devenderbutani21** [comment]
Okay! I will make smaller PRs for easier merge. Thanks for the heads up.

**Comment by TreyWW** [comment]
> @TreyWW Thanks for the help. You mentioned blank = True at the top. In cases like the Invoice class in models.py why is blank = True even necessary? How does the functionality change if I just remove it from each variable and leave null = True to dictate what needs to be in those fields? Are all these fields not supposed to be mandatory?

Hey @vishalprashant,

Blank=True just fixes the admin page, otherwise it makes every field required. Any field with null we should set with blank

**Comment by Yugalgarg2002** [comment]
I also want to contribute and start my open source contribution journey.

**Comment by TreyWW** [comment]
Hey @Yugalgarg2002, welcome along! Let me know if you have any questions

**Comment by brainycarp67531** [comment]
Hi @TreyWW, I would like to start contributing to the cleaning of the code.

**Comment by TreyWW** [comment]
Hey @brainycarp67531, welcome to the project! Sure thing, what parts would you like to work on? Django Templates/HTML, Python, Service layer (would be helpful to have better structural ideas), developer documentation, anything like that in specific?

Thanks!

**Comment by brainycarp67531** [comment]
@TreyWW, I would like to start with python, but open to other areas also.

**Comment by TreyWW** [comment]
Sure thing. One core part of our new architecture is the Service layer. This sits between the backend request, and the response. I'd appreciate a second set of ideas to approach this problem, as I find my ways of doing things are a little... odd. 

 Give me a few minutes and I'll gather some more detail for you. Appreciate it!

**Comment by TreyWW** [comment]
@brainycarp67531 i've created #442, I'll post the details there

**Comment by shubacca** [comment]
hi @TreyWW 
I wanted to add in this bit of docstrings to the currency_conversion module (backend\api\currency_converter\convert.py). please check if this is ok? this is my first time contributing to open source:
def currency_conversion(request: HtmxHttpRequest):
    """
    Handles incoming Django HTTP requests, checks for rate limits, validates input, calls     convert_currency function and returns the appropriate response.

    Parameters
    ----------
    request: HtmxHttpRequest, required
        Django HTTP request object

    Returns
    ----------
    Returns the correct rendering of the expected response
    """

**Comment by TreyWW** [comment]
Hi @shubacca,

We are currently removing currency conversion (an open PR) so it would be useless to add them. Apologies!

We have some open code that can be revamped though, feel free to look at previous comments or I can find you something :) 

Thanks for your interest!

**Comment by shubacca** [comment]
Thanks for the quick reply, no worries. I can look at the comments above, and look for a particular issue. In the meantime, if you do have something pertinent along this issue's lines, let me know and I am open!

**Comment by TreyWW** [comment]
Amazing thanks @shubacca. I'm on holiday at the moment but when I get back I can find something if youre still available. I'll keep in mind :)

**Comment by Kenobi767** [comment]
ready to contribute to code clean up. Which (if any) still need doing?

**Comment by TreyWW** [comment]
Hey @Kenobi767 apologies for the late response. Models.py needs a few refactors as discussed in the thread. Blank=True, function names, maybe even separation if appropriate

Let me know if you'd like any more info. Feel free to create a branch, then create a draft PR right away and we can discuss changes as you go. 

Thanks for your interest!

**Comment by Kenobi767** [comment]
I'm having problems with the installation, when I run manage.py migrate I get an error message saying that storages cannot be imported. I know its my end, never done anything like this before. and haven't managed to install it with pip or poetry. Any debug advice?

  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "D:\github\MyFinances\settings\settings.py", line 14, in <module>
    from storages.backends.s3 import S3Storage
ModuleNotFoundError: No module named 'storages'

**Comment by TreyWW** [comment]
Hi @Kenobi767,

Do you have discord by any chance? Slightly easier to have a conversation there. If not let me know that's absolutely fine. Ill try and debug on my end :)

**Comment by TreyWW** [comment]
But yeah you seem to be correct about the poetry bit.

On the docs [here](https://docs.myfinances.cloud/getting-started/installation/#setup-the-backend-django) you'll need to make sure you're in the venv, then run

```
pip install poetry

poetry install --no-root --with mypy,django,dev
```

Let me know the errors if you get any with poetry. Feel free to create a new issue as "Bug" or join the discord if you would like to continue the discussion :)

**Comment by Geary-E** [comment]
I'm definitely interested. Does models.py still need to be refactored?

**Comment by LaraMerdol** [comment]
Hello, we are a group of university students working on an open-source contribution assignment. We'd like to work on this issue—can it be assigned to me?
